<a href="https://colab.research.google.com/github/jhonataneduardoroagonzalez-debug/Proyectos_Python/blob/main/Estad%C3%ADstica_Descriptiva_jr.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install rpy2 # Con este código se puede trabajar simultaneamente el lenguaje python y el lenguaje R
%load_ext rpy2.ipython

The rpy2.ipython extension is already loaded. To reload it, use:
  %reload_ext rpy2.ipython


In [ ]:
# Importando liberias de python
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import math
from scipy import stats

In [ ]:
from google.colab import drive
drive.mount('/content/drive')
df = pd.read_excel('/content/Prostata.xlsx')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
#from google.colab import files
#df = pd.read_excel('/content/Prostata.xlsx')

Para el análisis que se presenta a continuación, se utiliza un data frame que contiene 6 variables con 97 registros de hombres  con cáncer  de  próstata  que  debían  someterse  a  una  prostatectomía  radical.    

In [ ]:
# Importando los datos
df_prostata=pd.read_excel('Prostata.xlsx')
df['weight'] = df['weight'].round(2)
df.head()

cavol  weight  age   bph  svi
0   0.56   15.95   50  0.25    0
1   0.37   27.65   58  0.25    0
2   0.60   14.75   74  0.25    0
3   0.30   26.65   58  0.25    0
4   2.12   30.95   62  0.25    0

## **Distribución de Frecuencias**

A continuación, se presenta la construcción de una distribución de frecuencias para cada una de las  variaibles que componen el dataframe, inicialmente.

En primera instancia se presenta la frecuencia $\displaystyle{n_{i}}$ la cual corresponde al conteo o número de veces que aparece el valor o el atributo de la variable

### **Número de integrantes del hogar**

In [ ]:
# Distribución de frecuencias
(df
 .groupby(['cavol', 'weight', 'age', 'bph', 'svi'])
 .size()
 .to_frame(name='frequency')
)

frequency
cavol     weight    age bph      svi           
0.260000  36.550688 63  3.550001 0            1
0.300000  26.650289 58  0.250000 0            1
0.350000  25.249340 50  0.250000 0            1
0.370000  27.649289 58  0.250000 0            1
0.460000  34.449690 47  0.250000 0            1
...                                         ...
23.420001 33.649561 59  0.250000 0            1
25.700000 60.448998 68  0.250000 0            1
27.190001 33.750661 64  0.250000 1            1
32.200002 53.250117 68  1.550000 1            1
45.650000 49.249539 44  0.250000 1            1

[97 rows x 1 columns]

In [ ]:
# Distribución de frecuencias de 'Cavol' agregando las correspondientes frecuencias relativa, porcentual y acumulada

# Calculate descriptive statistics for the original 'cavol' column
print("Máximo:",df['cavol'].max())
print("Mínimo:",df['cavol'].min())
print("Rango:",(df['cavol'].max()-df['cavol'].min()))
print("Número de Intervalos:",math.ceil(1+(3.3*np.log10(len(df['cavol'])))))

# Create intervals and frequency table
df["cavol_group"] = pd.cut(df["cavol"],bins=8) # Using 8 bins as in cell ovwpa4DZCG2F
frequency0 = (df
 .groupby('cavol_group', observed=True)
 .agg(frequency=('cavol_group','count'))
)

# Calculate relative frequency
frequency0['relative_frequency'] = frequency0['frequency'] / frequency0['frequency'].sum()

frequency0['percentage_frequency'] = frequency0['relative_frequency'].round(3)*100
frequency0['cumulative_frequency'] = frequency0['frequency'].cumsum()
frequency0['cumulative_relative_frequency'] = frequency0['relative_frequency'].cumsum().round(3)
frequency0['cumulative_percentage_frequency'] = frequency0['percentage_frequency'].cumsum().round(3)
display(frequency0)

Máximo: 45.649999678579384
Mínimo: 0.2600000124713187
Rango: 45.38999966610807
Número de Intervalos: 8


frequency  relative_frequency  percentage_frequency  \
cavol_group                                                             
(0.215, 5.934]           62            0.639175                  63.9   
(5.934, 11.607]          14            0.144330                  14.4   
(11.607, 17.281]         12            0.123711                  12.4   
(17.281, 22.955]          3            0.030928                   3.1   
(22.955, 28.629]          4            0.041237                   4.1   
(28.629, 34.302]          1            0.010309                   1.0   
(39.976, 45.65]           1            0.010309                   1.0   

                  cumulative_frequency  cumulative_relative_frequency  \
cavol_group                                                             
(0.215, 5.934]                      62                          0.639   
(5.934, 11.607]                     76                          0.784   
(11.607, 17.281]                    88                          0.907   
(17.281, 22.955]                    91                          0.938   
(22.955, 28.629]                    95                          0.979   
(28.629, 34.302]                    96                          0.990   
(39.976, 45.65]                     97                          1.000   

                  cumulative_percentage_frequency  
cavol_group                                        
(0.215, 5.934]                               63.9  
(5.934, 11.607]                              78.3  
(11.607, 17.281]                             90.7  
(17.281, 22.955]                             93.8  
(22.955, 28.629]                             97.9  
(28.629, 34.302]                             98.9  
(39.976, 45.65]                              99.9

El volumen mas alto de cancer de prostata es de 45.65 y el mas pequeño es de 0.26.




In [ ]:
# Distribución de frecuencias para 'age'. frecuencias relativa, porcentual y acumulada
frequency1 = (df
 .groupby('age')
 .agg(frequency=('age','count')))
frequency1['relative_frequency'] = frequency1['frequency'] / frequency1['frequency'].sum()
frequency1['percentage_frequency'] = frequency1['relative_frequency']*100
frequency1['cumulative_frequency'] = frequency1['frequency'].cumsum()
frequency1['cumulative_relative_frequency'] = frequency1['relative_frequency'].cumsum()
frequency1['cumulative_percentage_frequency'] = frequency1['percentage_frequency'].cumsum().round(3)
display(frequency1)


frequency  relative_frequency  percentage_frequency  \
age                                                        
41           1            0.010309              1.030928   
43           1            0.010309              1.030928   
44           1            0.010309              1.030928   
47           2            0.020619              2.061856   
49           1            0.010309              1.030928   
50           2            0.020619              2.061856   
52           1            0.010309              1.030928   
54           1            0.010309              1.030928   
56           1            0.010309              1.030928   
57           2            0.020619              2.061856   
58           4            0.041237              4.123711   
59           3            0.030928              3.092784   
60           5            0.051546              5.154639   
61           5            0.051546              5.154639   
62           3            0.030928              3.092784   
63           7            0.072165              7.216495   
64           7            0.072165              7.216495   
65           7            0.072165              7.216495   
66           6            0.061856              6.185567   
67           3            0.030928              3.092784   
68          12            0.123711             12.371134   
69           5            0.051546              5.154639   
70           4            0.041237              4.123711   
71           1            0.010309              1.030928   
72           4            0.041237              4.123711   
73           2            0.020619              2.061856   
74           1            0.010309              1.030928   
76           1            0.010309              1.030928   
77           2            0.020619              2.061856   
78           1            0.010309              1.030928   
79           1            0.010309              1.030928   

     cumulative_frequency  cumulative_relative_frequency  \
age                                                        
41                      1                       0.010309   
43                      2                       0.020619   
44                      3                       0.030928   
47                      5                       0.051546   
49                      6                       0.061856   
50                      8                       0.082474   
52                      9                       0.092784   
54                     10                       0.103093   
56                     11                       0.113402   
57                     13                       0.134021   
58                     17                       0.175258   
59                     20                       0.206186   
60                     25                       0.257732   
61                     30                       0.309278   
62                     33                       0.340206   
63                     40                       0.412371   
64                     47                       0.484536   
65                     54                       0.556701   
66                     60                       0.618557   
67                     63                       0.649485   
68                     75                       0.773196   
69                     80                       0.824742   
70                     84                       0.865979   
71                     85                       0.876289   
72                     89                       0.917526   
73                     91                       0.938144   
74                     92                       0.948454   
76                     93                       0.958763   
77                     95                       0.979381   
78                     96                       0.989691   
79                     97                       1.000000   

     cumulative_percentage_frequency  
age 

In [ ]:
%%R
ggplot(age, aes(x = "", fill = factor(estrato))) +
  geom_bar(width = 1) +
  coord_polar("y", start = 0) +
  labs(fill = "Estrato") +
  theme_void()

A los 68 años se tiene el mayor porcentaje con un 75% (frecuencia acomulada) de casos de prostatectomia con la frecuencia mas alta de 12

In [ ]:
# Distribución de frecuencias para 'weight' : frecuencias relativa, porcentual y acumulada
frequency2 = (df
 .groupby('weight')
 .agg(frequency2=('weight','count'))).round(4)

frequency2['relative_frequency'] = frequency2['frequency2'] / frequency2['frequency2'].sum().round(5)
frequency2['percentage_frequency'] = frequency2['relative_frequency'].round(3)*100
frequency2['cumulative_frequency'] = frequency2['frequency2'].cumsum()
frequency2['cumulative_relative_frequency'] = frequency2['relative_frequency'].cumsum().round(4)
frequency2['cumulative_percentage_frequency'] = frequency2['percentage_frequency'].cumsum().round(2)
display(frequency2)

# por intervalo
import math
print("Máximo:",round(df['weight'].max(), 2))
print("Mínimo:",round(df['weight'].min(), 2))
print("Rango:",round(df['weight'].max()-df['weight'].min(), 2))
print("Número de intervalos:",math.ceil(1+(3.3*np.log10(len(df['weight'])))))
# Amplitud de los intervalos

frequency2  relative_frequency  percentage_frequency  \
weight                                                             
10.749938            1            0.010309                   1.0   
14.749365            1            0.010309                   1.0   
15.950657            1            0.010309                   1.0   
17.550808            1            0.010309                   1.0   
20.049415            1            0.010309                   1.0   
...                ...                 ...                   ...   
84.250963            1            0.010309                   1.0   
92.249789            1            0.010309                   1.0   
111.955335           1            0.010309                   1.0   
119.152001           1            0.010309                   1.0   
449.259198           1            0.010309                   1.0   

            cumulative_frequency  cumulative_relative_frequency  \
weight                                                            
10.749938                      1                         0.0103   
14.749365                      2                         0.0206   
15.950657                      3                         0.0309   
17.550808                      4                         0.0412   
20.049415                      5                         0.0515   
...                          ...                            ...   
84.250963                     93                         0.9588   
92.249789                     94                         0.9691   
111.955335                    95                         0.9794   
119.152001                    96                         0.9897   
449.259198                    97                         1.0000   

            cumulative_percentage_frequency  
weight                                       
10.749938                               1.0  
14.749365                               2.0  
15.950657                               3.0  
17.550808                               4.0  
20.049415                               5.0  
...                                     ...  
84.250963                              93.9  
92.249789                              94.9  
111.955335                             95.9  
119.152001                             96.9  
449.259198                             97.9  

[88 rows x 6 columns]

Máximo: 449.26
Mínimo: 10.75
Rango: 438.51
Número de intervalos: 8


El tumor más grande pesa 449.26 gramos mientras que el más pequeño es de 10.75 gramos.

In [ ]:
# Distribución de frecuencias para 'bph': frecuencias relativa, porcentual y acumulada
frequency3 = (df
 .groupby('bph')
 .agg(frequency3=('bph','count')))

frequency3['relative_frequency'] = frequency3['frequency3'] / frequency3['frequency3'].sum()
frequency3['percentage_frequency'] = frequency3['relative_frequency'].round(2)*100
frequency3['cumulative_frequency'] = frequency3['frequency3'].cumsum()
frequency3['cumulative_relative_frequency'] = frequency3['relative_frequency'].cumsum().round(2)
frequency3['cumulative_percentage_frequency'] = frequency3['percentage_frequency'].cumsum().round(2)
display(frequency3)

# por intervalo
import math
print("Máximo:",round(df['bph'].max(), 2))
print("Mínimo:",round(df['bph'].min(), 2))
print("Rango:",round(df['bph'].max()-df['bph'].min(), 2))
print("Número de intervalos:",math.ceil(1+(3.3*np.log10(len(df['bph'])))))
# Amplitud de los intervalos

frequency3  relative_frequency  percentage_frequency  \
bph                                                               
0.250000           43            0.443299                  44.0   
0.450000            1            0.010309                   1.0   
0.560000            1            0.010309                   1.0   
0.590000            1            0.010309                   1.0   
0.950000            1            0.010309                   1.0   
1.250001            1            0.010309                   1.0   
1.350001            1            0.010309                   1.0   
1.550000            5            0.051546                   5.0   
1.750000            1            0.010309                   1.0   
1.850001            2            0.020619                   2.0   
2.549999            2            0.020619                   2.0   
3.050001            1            0.010309                   1.0   
3.470001            1            0.010309                   1.0   
3.550001            2            0.020619                   2.0   
3.649999            1            0.010309                   1.0   
3.750001            1            0.010309                   1.0   
3.849999            1            0.010309                   1.0   
3.950002            2            0.020619                   2.0   
4.149999            1            0.010309                   1.0   
4.370000            1            0.010309                   1.0   
4.450000            1            0.010309                   1.0   
4.649999            1            0.010309                   1.0   
4.750002            1            0.010309                   1.0   
4.950002            1            0.010309                   1.0   
5.049999            1            0.010309                   1.0   
5.150001            1            0.010309                   1.0   
5.250000            1            0.010309                   1.0   
5.450002            1            0.010309                   1.0   
5.499999            1            0.010309                   1.0   
5.550000            2            0.020619                   2.0   
5.750001            2            0.020619                   2.0   
5.850002            1            0.010309                   1.0   
5.949999            1            0.010309                   1.0   
6.149999            1            0.010309                   1.0   
6.550000            1            0.010309                   1.0   
7.120002            1            0.010309                   1.0   
7.450000            1            0.010309                   1.0   
7.950001            1            0.010309                   1.0   
8.350004            2            0.020619                   2.0   
8.770002            1            0.010309                   1.0   
10.050004           1            0.010309                   1.0   
10.240004           3            0.030928                   3.0   

           cumulative_frequency  cumulative_relative_frequency  \
bph                                                              
0.250000                     43                           0.44   
0.450000                     44                           0.45   
0.560000                     45                           0.46   
0.590000                     46                           0.47   
0.950000                     47                           0.48   
1.250001                     48                           0.49   
1.350001                     49                           0.51   
1.550000                     54                           0.56   
1.750000                     55                           0.57   
1.850001                     57                           0.59   
2.549999                     59                           0.61   
3.050001                     60                           0.62   
3.470001                     61                           0.63   
3.550001                     63                           0.65   
3.6499

Máximo: 10.24
Mínimo: 0.25
Rango: 9.99
Número de intervalos: 8


El crecimiento excesivo no cacerigeno en los hombres de la base de datos no supera los 10.24 gramos

In [ ]:
# Distribución de frecuenciasopara 'bph'. frecuencias relativa, porcentual y acumulada
frequency4 = (df
 .groupby('svi')
 .agg(frequency4=('svi','count')))

frequency4['relative_frequency'] = frequency4['frequency4'] / frequency4['frequency4'].sum()
frequency4['percentage_frequency'] = frequency4['relative_frequency'].round(2)* 100
frequency4['cumulative_frequency'] = frequency4['frequency4'].cumsum()
frequency4['cumulative_relative_frequency'] = frequency4['relative_frequency'].cumsum().round(2)
frequency4['cumulative_percentage_frequency'] = frequency4['percentage_frequency'].cumsum().round(2)
display(frequency4)

frequency4  relative_frequency  percentage_frequency  \
svi                                                         
0            76            0.783505                  78.0   
1            21            0.216495                  22.0   

     cumulative_frequency  cumulative_relative_frequency  \
svi                                                        
0                      76                           0.78   
1                      97                           1.00   

     cumulative_percentage_frequency  
svi                                   
0                               78.0  
1                              100.0

El 21.65% de los hombres con cancer tienen una invación de la vesicula seminal.

In [ ]:
s = df["cavol"].astype(float)
bins = [0, 0.5, 2, 10, float("inf")]
labels = ["0–0.5", "0.5–2", "2–10", "≥10"]

cats = pd.cut(s, bins=bins, labels=labels, include_lowest=True, right=True)
res = cats.value_counts().sort_index()
pct = (res / len(s) * 100).round(1)

In [ ]:
ax = res.plot(kind="bar", color=["#7fc97f","#beed6f","#fdc086","#e31a1c"], edgecolor="black", figsize=(8,5))
ax.set_title("Distribución cavol por bloques")
ax.set_ylabel("Número de casos")
ax.set_xlabel("Bloque (cm³)")
ax.grid(axis="y", linestyle=":", alpha=0.4)
total = int(res.sum())
for p, v in zip(ax.patches, res.values):
    ax.annotate(f"{v} ({v/total*100:.1f}%)", (p.get_x()+p.get_width()/2, v), ha="center", va="bottom")

plt.tight_layout()
plt.show()

ANALISIS

Se hizo una distribucion de los datos de cavol por bloques. Donde eje x muestra 4 categorias de rangos. Don la clasificacion representa:

El gráfico muestra la distribución de los valores de cavol agrupados en cuatro rangos. Se observa que el 48,5 % de los casos se concentran en el rango 2–10, lo que sugiere que casi la mitad de los pacientes presentan tumores en una etapa avanzada, requiriendo una intervención médica inmediata, como procedimientos quirúrgicos o tratamientos intensivos.

Por otra parte, los grupos comprendidos entre 0,5–2 (23,7 %) y ≥ 10 (22,7 %) representan en conjunto cerca del 45 % de la población analizada, lo que indica un número considerable de pacientes con tumores que requieren seguimiento clínico y tratamientos de control para evitar su expansión.

Finalmente, el 5,2 % restante, correspondiente al rango 0–0,5, refleja los casos con mínimos indicios de actividad tumoral o sin presencia de tumores malignos significativos, lo que sugiere un bajo nivel de riesgo en este grupo.

In [ ]:
modes = [round(float(x), 6) for x in s.mode().tolist()]
res_df = pd.DataFrame({
    "valor": {
        "n": int(s.shape[0]),
        "min": s.min(),
        "max": s.max(),
        "media": s.mean(),
        "mediana": s.median(),
        "moda": ", ".join(map(str, modes)),  # ya redondeada
        "varianza_poblacional": s.var(ddof=0),
        "varianza_muestral": s.var(ddof=1),
        "desv_std_poblacional": s.std(ddof=0),
        "desv_std_muestral": s.std(ddof=1),
        "q1": s.quantile(0.25),
        "q2_mediana": s.quantile(0.50),
        "q3": s.quantile(0.75),
        "rango_intercuartilico": s.quantile(0.75) - s.quantile(0.25),
        "rango": s.max() - s.min(),
        "sesgo": stats.skew(s, bias=False),
        "curtosis_exceso": stats.kurtosis(s, fisher=True, bias=False),
        "curtosis_total": stats.kurtosis(s, fisher=True, bias=False) + 3,
    }
})
# Redondear resultados a 3 decimales
res_df_rounded = res_df.round(3)

# Configurar pandas para mostrar solo 3 decimales
pd.set_option('display.precision', 3)
print(res_df_rounded)

ANALISIS

Al relacionar la formula con el grafico, vemos que al estar en cuatro bloques, no representa que el valor minimo esta de* 0-0.5 donde nuestros datos nos muestran el valor minimo que es de 0.26 cm³. donde nos muestra que este rango suelen corresponder a lesiones muy pequeñas-y el mavlor maximo de nuestro datos es de 45.65 cm³ donde nos muestra que es el caso con mayor carga muzcular. Tambien podemos ver que la media de nuestros datos es de 7.00 cm³ es el promedio de todos los valores.podemos ver que en nuestros datos son sencibles a valores altos ya que la mayoria estan de ≥10 cm³ (y hasta ~45.65),* que “empujan” la media hacia arriba. luego podemos ver que en nuestros datos de la mediana que es el dato central es de 4.25 cm³, lo que podemos identificar es que la mediana cae en el bloque 2–10 cm³ esto es coherente con que la barra mas alta sea justo eso con el 48.5%cm³ de los casos. esto quiere decir que este dato nos da el tamaño tipo del tumor de nuestros datos. continuamos con la moda que nos indica los volumenes especificos que se repiten pero como vemos nuestros datos son conitinuos y suele ser inestable y poco representativos del conjunto.Aunque algunas modas estén en* 0.5–2* y otra en 2–10, el gráfico no muestra valores puntuales, solo conteos por rangos. La utilidad de la moda aquí es limitada.

Lavarianza poblacuional de cavol es de 61.54 (cm³)², lo que refleja una alta variabilidad entre pacientes, impulsada por un subconjunto de tumores de gran volumen (≥10 cm³).

La varianzaza muestral fue 62.18 (cm³)² y la desviación estándar muestral 7.89cm³evidenciando alta heterogeneidad del volumen tumoral. Esta dispersión está impulsada por un subconjunto de lesiones de gran volumen (≥10 cm³), visibles en la distribución por bloques, que elevan la media (7.00 cm³) por encima de la mediana (4.25 cm³).

Los cuartiles de cavol fueron Q1=1.67 cm³, mediana=4.25 cm³ y Q3=8.39 cm³. El 50% central de pacientes se ubica entre 1.67 y 8.39 cm³ (IQR=6.72 cm³),indicando un tamaño tipico intermedio, con una fraccion no despreciable en la cola superior que supera los cm³.

Tambien podemos ver que El IQR fue 6.72 cm³ (Q1=1.67; Q3=8.39), indicando que el 50% central se concentra entre 1.67 y 8.39 cm³, dentro del bloque 2–10 cm³ . El rango total fue 45.39 cm³ (0.26 a 45.65), evidenciando la presencia de casos muy voluminosos. El sesgo fue positivo (2.18), mostrando una asimetría hacia volúmenes altos, consistente con la fracción de casos en ≥10 cm³ que eleva la media por encima de la mediana

como ultimo El exceso de curtosis fue 6.13* (curtosis total 9.13)*, lo que indica colas muy pesadas respecto de la normal. En cavol, esto se traduce en una proporción apreciable de volúmenes altos y valores extremos (≥10 cm³, algunos >20–45 cm³) que incrementan la probabilidad de observar tumores de gran tamaño.

In [ ]:

# --- Clasificación según el peso de la próstata ---
def clasificar_peso(peso):
    if 0 <= peso < 30:
        return "Normal (0 - 30 gr)"
    elif 30 <= peso < 50:
        return "Levemente aumentada (30 - 50 gr)"
    elif 50 <= peso < 80:
        return "Moderadamente aumentada (50 - 80 gr)"
    elif 80 <= peso < 100:
        return "Muy aumentada / masiva (80 - 100 gr)"
    elif 100 <= peso < 500:
        return "Información a Revisar (> 100 gr)"
    else:
        return "Fuera de rango"

# Crear columna con la clasificación
df_Prostata["weight classification"] = df_Prostata["weight"].apply(clasificar_peso)

# Calcular frecuencias
frecuencia_peso = (
    df_Prostata
    .groupby("weight classification", observed=True)
    .agg(frequency=('weight classification', 'count'))
    .reset_index()
)

# Ordenar las categorías de forma lógica
orden = ["Normal (0 - 30 gr)", "Levemente aumentada (30 - 50 gr)", "Moderadamente aumentada (50 - 80 gr)", "Muy aumentada / masiva (80 - 100 gr)", "Información a Revisar (> 100 gr)"]
frecuencia_peso["weight classification"] = pd.Categorical(frecuencia_peso["weight classification"], categories=orden, ordered=True)
frecuencia_peso = frecuencia_peso.sort_values("weight classification")

# Colores personalizados
colors = ["#4CAF50", "#FFEB3B", "#FF9800", "#F44336", "#000000"]  # verde, amarillo, naranja, rojo, negro

# Crear gráfico de anillo
fig, ax = plt.subplots(figsize=(6,6))
wedges, texts, autotexts = ax.pie(
    frecuencia_peso["frequency"],
    labels=frecuencia_peso["weight classification"],
    colors=colors,
    startangle=90,
    autopct=lambda pct: f"{int(round(pct * frecuencia_peso['frequency'].sum() / 100))}",  # solo conteo
    wedgeprops=dict(width=0.35)
)

# Ajustes estéticos
plt.setp(autotexts, size=10, weight="bold", color="black")
ax.set_title("Clasificación del peso prostático", fontsize=14, fontweight='bold')
ax.set(aspect="equal")


**ANÁLISIS**

De acuerdo al comportamiento de la clasificación de peso prostático y argumentados en *Does the Size Matter? Prostate Weight Does Not Predict PSA* (Para identificar valores o parámetros sobre el estado del mismo) identificamos:

*   El **28,86%** de la población estudiada presenta un peso prostático normal, oscilado a lo sumo con 30 gramos. Los pcaientes deben recurrir a exámenes periódicos para tener la certeza de que el pronóstico no presente alteraciones con el pasar del tiempo
*   El **68,04%** de la población presenta un peso prostático por fuera de los parámetros normales. Del **68,4%**; el **49,4%** presente un peso prostático levemente aumentado, puede que presente hiperplasia benigna, los pacientes en mención presentan síntomas como flujo urinario débil o intermitente, sensación de vaciado incompleto de la vejiga y/o aumento de la frecuencia urinaria, para ellos es importante la evaluación de posibles tratamientos médicos y/o urinarios. El **15,46%** de la población presenta peso moderadamente aumentada, para estos pacientes pueden tener síntomas síntomas urinarios, obligatoriamente debe recurrrir a tratamiento médico para prevenir complicaciones y reducir volumen prostático. Para el **3,09%** restante los pacientes presentan tamaño porstático muy aumentado, lo que refiere a alto riesgo de tumor voluminoso o hiperplasia avanzada, para ello debe recurrir a tratamiento farmacológico o intervención quirúrgica urgente, dado que tiene alta probabilidad de presentar metástatis.

* El **3,09%** correspondiente a *información a revisar* se debe a datos que consideramos como equipo de trabajo atípicos, dado que hay datos mayores a 100 gr, lo cual, de acuerdo a la teoría revisada no es posible que se puedan tener los valores en mención.

In [ ]:


# Calcular medidas estadísticas
media = df['weight'].mean()
mediana = df['weight'].median()
moda = df['weight'].mode()[0]
desviacion_estandar = df['weight'].std()
rango = df_Prostata['weight'].max() - df_Prostata['weight'].min()
rango_intercuartil = df_P['weight'].quantile(0.75) - df_Prostata['weight'].quantile(0.25)


# Detectar valores atípicos mayores a 100
atipicos = df_Prostata[df_Prostata['weight'] > 100]

# Crear gráfico tipo boxplot
plt.figure(figsize=(10, 6))
sns.boxplot(x=df_Prostata['weight'], color="#AED6F1", width=0.4)

# === Marcar líneas de medidas ===
plt.axvline(media, color='red', linestyle='--', label=f'Media: {media:.1f} g')
plt.axvline(mediana, color='green', linestyle='-', label=f'Mediana: {mediana:.1f} g')
plt.axvline(moda, color='orange', linestyle='-.', label=f'Moda: {moda:.1f} g')


# === Agregar anotaciones de texto ===
plt.text(media + 0.5, 0.1, f'Media\n{media:.1f} g', color='red', fontsize=10, rotation=90, va='bottom')
plt.text(mediana + 0.5, 0.1, f'Mediana\n{mediana:.1f} g', color='green', fontsize=10, rotation=90, va='bottom')
plt.text(moda + 0.5, 0.1, f'Moda\n{moda:.1f} g', color='orange', fontsize=10, rotation=90, va='bottom')

# === Marcar valores atípicos ===
plt.scatter(atipicos['weight'], [0.05]*len(atipicos), color='black', s=80, marker='*', label='Atípicos > 100 g')

# Etiquetas y título
plt.title("Distribución del Peso Prostático (g)", fontsize=14, fontweight="bold")
plt.xlabel("Peso de la próstata (gramos)")
plt.legend()
plt.grid(axis='x', linestyle=':', alpha=0.7)

# Rango interquartilico
Q1 = df_Prostata['weight'].quantile(0.25)
Q3 = df_Prostata['weight'].quantile(0.75)
IQR = Q3 - Q1
print("Rango intercuartílico (IQR):", IQR)

# Mostrar gráfico
plt.show()

**Análisis**
*   El peso Prostático tiene una media de **45,5 gr** y una mediana de **37,4 gr**. De acuerdo al análisis anterior. La media y mediana  se concentran entre peso próstatico levemente aumentada. La moda se concentra en **26,3 gr**, dado que existen datos consistentes alrededor del comportamiento normal, lo que refiere a que existen pacientes que se encuentran próximos a pasar a la etapa *Levemente aumentada*, en ellos se debe generar una "alerta" para tomar acciones preventivas y así generar un hábito responsable sobre el seguimiento constante a evitar el cáncer de próstata.
* Existen gráficos por fuera de los cuartiles, los cuales corresponden a valores mayores a 100 gr. La data genera cierta incertidumbre sobre su validez dado que el valor mayor debe ser de 100 gr en su estado más crítico.

In [ ]:
plt.figure(figsize=(8, 5))

# Crear el histograma con Seaborn (más estético que solo Matplotlib)
sns.histplot(df_Prostata['age'], bins=8, kde=True, color='skyblue', edgecolor='black')

# Agregar etiquetas con las frecuencias sobre cada barra
for patch in ax.patches:
    height = patch.get_height()
    if height > 0:  # Evitar etiquetas sobre barras vacías
        ax.text(
            patch.get_x() + patch.get_width() / 2,  # posición x centrada
            height + 0.5,  # posición y justo encima de la barra
            f'{int(height)}',  # texto (frecuencia redondeada)
            ha='center', va='bottom', fontsize=10, color='black'
        )
# Títulos y etiquetas
plt.title("Histograma de Edad de Pacientes con Cáncer de Próstata", fontsize=13)
plt.xlabel("Edad")
plt.ylabel("Frecuencia")
plt.legend()
plt.grid(axis='y', linestyle='--', alpha=0.6)

plt.show()

In [ ]:
a = df_Prostata['age']


# Crear DataFrame con todas las medidas estadísticas
res_df = pd.DataFrame({
    "valor": {
        "n": int(a.shape[0]),
        "min": a.min(),
        "max": a.max(),
        "media": a.mean(),
        "mediana": a.median(),
        "moda": ", ".join(map(str, modes)),
        "varianza": a.var(ddof=1),
        "desviación": a.std(ddof=1),
        "q1": a.quantile(0.25),
        "q2_mediana": a.quantile(0.50),
        "q3": a.quantile(0.75),
        "rango_intercuartilico": a.quantile(0.75) - a.quantile(0.25),
        "rango": a.max() - a.min(),
        "sesgo": stats.skew(a, bias=False),
        "curtosis_total": stats.kurtosis(a, fisher=True, bias=False) + 3,
    }
})

# Redondear resultados a 3 decimales
res_df_rounded = res_df.round(3)

# Configurar pandas para mostrar solo 3 decimales
pd.set_option('display.precision', 3)

print("\n📊 Estadísticos descriptivos de la variable 'age':")
print(res_df_rounded)

In [ ]:
# Datos
datos = df_Prostata['bph']

# Crear el histograma y guardar las frecuencias y bordes de los intervalos
frecuencias, bordes, _ = plt.hist(datos, bins=8, color='skyblue', edgecolor='black')

# Agregar títulos y etiquetas
plt.title('Histograma de Frecuencia - BPH')
plt.xlabel('Cantidad de Hiperplasia Prostática (gramos)')
plt.ylabel('Frecuencia')

# Poner los intervalos (bordes) en el eje X
plt.xticks(np.round(bordes, 2))  # redondea a 2 decimales para mostrar bonito

# Agregar etiquetas de frecuencia encima de cada barra
for i in range(len(frecuencias)):
    x = (bordes[i] + bordes[i+1]) / 2  # centro de cada barra
    y = frecuencias[i]
    plt.text(x, y + 0.5, int(y), ha='center', va='bottom', fontsize=9, color='black')

# Mostrar el gráfico
plt.show()

In [ ]:
# Definir la serie de datos
b = df_Prostata['bph']

# Crear DataFrame con todas las medidas estadísticas
res_df = pd.DataFrame({
    "valor": {
        "n": int(b.shape[0]),
        "min": b.min(),
        "max": b.max(),
        "media": b.mean(),
        "mediana": b.median(),
        "moda": ", ".join(map(str, modes)),
        "varianza": b.var(ddof=1),
        "desviación": b.std(ddof=1),
        "q1": b.quantile(0.25),
        "q2_mediana": b.quantile(0.50),
        "q3": b.quantile(0.75),
        "rango_intercuartilico": b.quantile(0.75) - b.quantile(0.25),
        "rango": b.max() - b.min(),
        "sesgo": stats.skew(b, bias=False),
        "curtosis": stats.kurtosis(b, fisher=True, bias=False) + 3,
    }
})

# Redondear resultados a 3 decimales
res_df_rounded = res_df.round(3)

# Configurar pandas para mostrar solo 3 decimales
pd.set_option('display.precision', 3)

print("\n📊 Estadísticos descriptivos de la variable 'bph':")
print(res_df_rounded)

El 50,5% de los hombres con cáncer de próstata presenta una cantidad baja de hiperplasia prostática benigna, con valores entre 0,25 y 1,5 gramos. En general, los valores de BPH se concentran en rangos bajos, mostrando que la presencia de hiperplasia benigna es menor dentro de la muestra. Sin embargo, existen algunos casos aislados con cantidades elevadas de tejido benigno superiores a 7 gramos que representa menos del 10% de la muestra total y generan una ligera dispersión en los resultados. Esto coincide con la naturaleza de la muestra, compuesta principalmente por hombres con diagnóstico de cáncer prostático que se asocia más con el tumor maligno que con procesos benignos.

### **SVI**

In [ ]:
# 1. Elige el nombre exacto de tu columna
nombre_columna_svi = 'svi'

# 2. Configura el tamaño del gráfico
plt.figure(figsize=(8, 6))

# 3. Crea el gráfico de barras con sns.countplot
ax = sns.countplot(
    x=nombre_columna_svi,
    data=df_Prostata,
    hue=nombre_columna_svi,  # <-- 1. Asigna la misma columna a 'hue'
    palette=['#3498db', '#e74c3c'],
    legend=False             # <-- 2. Oculta la leyenda para que el gráfico se vea igual
)

# 4. Añade títulos y etiquetas para que sea fácil de entender
ax.set_title(f'Distribución de la variable {nombre_columna_svi}', fontsize=16)
ax.set_xlabel('Respuesta', fontsize=12)
ax.set_ylabel('Frecuencia (Conteo)', fontsize=12)

# 5. (Opcional) Poner el número exacto encima de cada barra
for p in ax.patches:
    ax.annotate(f'{int(p.get_height())}',
                (p.get_x() + p.get_width() / 2., p.get_height()),
                ha='center', va='center',
                xytext=(0, 9),
                textcoords='offset points')

# 6. Muestra el gráfico
plt.show()

In [ ]:
media_svi = df_Prostata['svi'].mean()
moda_svi = df_Prostata['svi'].mode()[0]

print(f"La moda de SVI es: {moda_svi}")
print(f"La media de SVI (proporción de 'si') es: {media_svi:.2f}")
print(f"Esto significa que el {media_svi:.2%} de los casos tienen SVI SI.")

Los resultados obtenidos evidencian que el 78,3% de los hombres en estudio no presentan invasión. En contraste, el 21,65% restante corresponde a los individuos que sí registran presencia de invasión. Esta diferencia entre proporciones muestra una distribución desequilibrada, en la cual la ausencia de invasión representa la tendencia predominante.
El patrón observado sugiere que la variable SVI se caracteriza por una fuerte asimetría, con la mayoría de los valores agrupados en un extremo. Además, la dispersión general de los resultados es reducida, lo que refuerza la existencia de una tendencia central definida.